File: messages/src/main/java/com/thundashop/core/accountingmanager/SavedOrderFile.java
package com.thundashop.core.accountingmanager;

import com.thundashop.core.common.DataCommon;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.Date;
import java.util.HashMap;
import java.util.List;
import org.mongodb.morphia.annotations.Transient;

public class SavedOrderFile extends DataCommon {
    public List<AccountingTransaction> accountingTransactionLines = new ArrayList();
    public List<String> result;
    public String type = "accounting";
    public boolean transferred = false;
    public Date startDate;
    public Date endDate;
    public String subtype = ""; 
    public Double amountEx = 0.0;
    public Double amountInc = 0.0;
    public Double amountExDebet = 0.0;
    public Double amountIncDebet = 0.0;
    public Double sumAmountExOrderLines = 0.0;
    public Double sumAmountIncOrderLines = 0.0;
    public double onlyPositiveLinesEx = 0.0;
    public double onlyPositiveLinesInc = 0.0;
    
    public HashMap<String, Double> amountOnOrder = new HashMap();
    public List<String> tamperedOrders = new ArrayList();
    public List<String> orders = new ArrayList();
    public List<String> ordersTriedButFailed = new ArrayList();
    public String configId = "";
    public String transferId;
    public ArrayList ordersNow = new ArrayList();
    Date lastFinalized;
    public String base64Excel;
    
    public boolean needFinalize() {
        if(lastFinalized == null) {
            return true;
        }
        Calendar cal = Calendar.getInstance();
        int dayOfYear = cal.get(Calendar.DAY_OF_YEAR);
        cal.setTime(lastFinalized);
        int dayOfYear2 = cal.get(Calendar.DAY_OF_YEAR);
        return dayOfYear2 != dayOfYear;
    }
    
    @Transient
    public Integer numberOfOrdersNow = 0;
}

File: html5apps/terminal/events/core/accountingmanager/SavedOrderFile.php
<?php
class core_accountingmanager_SavedOrderFile extends core_common_DataCommon  {
	/** @var String */
	public $result;

	/** @var String */
	public $type;

	/** @var String */
	public $transferred;

	/** @var String */
	public $startDate;

	/** @var String */
	public $endDate;

	/** @var String */
	public $subtype;

	/** @var String */
	public $amountEx;

	/** @var String */
	public $amountInc;

	/** @var String */
	public $amountExDebet;

	/** @var String */
	public $amountIncDebet;

	/** @var String */
	public $sumAmountExOrderLines;

	/** @var String */
	public $sumAmountIncOrderLines;

	/** @var String */
	public $onlyPositiveLinesEx;

	/** @var String */
	public $onlyPositiveLinesInc;

	/** @var String */
	public $amountOnOrder;

	/** @var String */
	public $tamperedOrders;

	/** @var String */
	public $orders;

	/** @var String */
	public $ordersTriedButFailed;

	/** @var String */
	public $configId;

	/** @var String */
	public $transferId;

	/** @var String */
	public $ordersNow;

	/** @var String */
	public $lastFinalized;

	/** @var String */
	public $base64Excel;

	/** @var String */
	public $numberOfOrdersNow;

}
?>